src/libs/API.js

```diff
@@ -149,34 +167,7 @@ function handleExpiredAuthToken(originalCommand, originalParameters, originalTyp
  * @returns {Promise}
  */
 function request(command, parameters = {}, type = 'post') {
-    return new Promise((resolve, reject) => {
-        Network.post(command, parameters, type)
-            .then((response) => {
-                // Handle expired auth tokens properly by making sure to pass the resolve and reject down to the
-                // new promise created when calling handleExpiredAuthToken.
-                if (response.jsonCode === 407) {
-                    // Credentials haven't been initialized. We will not be able to re-authenticates with the API
-                    const unableToReauthenticate = (!credentials || !credentials.autoGeneratedLogin
-                        || !credentials.autoGeneratedPassword);
-
-                    // There are some API requests that should not be retried when there is an auth failure like
-                    // creating and deleting logins. In those cases, they should handle the original response instead
-                    // of the new response created by handleExpiredAuthToken.
-                    if (parameters.doNotRetry || unableToReauthenticate) {
-                        resolve(response);
-                        return;
-                    }
-
-                    handleExpiredAuthToken(command, parameters, type)
-                        .then(resolve)
-                        .catch(reject);
-                    return;
-                }
-
-                resolve(response);
-            })
-            .catch(reject);
-    });
+    return Network.post(command, parameters, type);
```

Comment:
If we are cool with these changes I would next just get rid of `request()` entirely.

Reply:
Wow, I love this. So awesome to see how that cleaned this up. I am all in favor of removing `request()` as well now.